components/caption_images/src/main.py

```diff
@@ -53,69 +52,39 @@ def caption_image_batch(
     return pd.Series(captions_batch, index=image_batch.index)
 
 
-def caption_images(
-    images: pd.Series,
-    *,
-    model: BlipForConditionalGeneration,
-    processor: BlipProcessor,
-    batch_size: int,
-    max_new_tokens: int,
-    device: str,
-) -> pd.DataFrame:
-    """Caption a pandas series of images."""
-    images = images.apply(process_image, processor=processor, device=device)
-    results: t.List[pd.Series] = []
-    for batch in np.split(images, np.arange(batch_size, len(images), batch_size)):
-        if not batch.empty:
-            results.append(
-                caption_image_batch(
-                    batch,
-                    model=model,
-                    processor=processor,
-                    max_new_tokens=max_new_tokens
-                ).T
-            )
-    return pd.concat(results).to_frame()
-
-
-class CaptionImagesComponent(DaskTransformComponent):
+class CaptionImagesComponent(PandasTransformComponent):
     """Component that captions images using a model from the Hugging Face hub."""
 
-    def transform(
-        self,
-        dataframe: dd.DataFrame,
-        model_id: str,
-        batch_size: int,
-        max_new_tokens: int,
-    ) -> dd.DataFrame:
-        """
-        Args:
-            dataframe: Dask dataframe
-            model_id: id of the model on the Hugging Face hub
-            batch_size: batch size to use
-            max_new_tokens: maximum token length of each caption.
-
-        Returns:
-            Dask dataframe
-        """
-        device = "cuda" if torch.cuda.is_available() else "cpu"
-        logger.info(f"Device: {device}")
-
-        processor = BlipProcessor.from_pretrained(model_id)
-        model = BlipForConditionalGeneration.from_pretrained(model_id)
-
-        dataframe = dataframe["images_data"].map_partitions(
-            caption_images,
-            model=model,
-            processor=processor,
-            batch_size=batch_size,
-            max_new_tokens=max_new_tokens,
-            device=device,
-            meta={0: str}
+    def setup(self, *, model_id: str, batch_size: int, max_new_tokens: int) -> None:
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
+        logger.info(f"Device: {self.device}")
+
+        self.processor = BlipProcessor.from_pretrained(model_id)
+        self.model = BlipForConditionalGeneration.from_pretrained(model_id)
```

Comment:
It doesn't seem like the model is put on the `device`

Reply:
Good catch. Seems like this wasn't done before either.